src/github.com/getlantern/flashlight/flashlight.go

```diff
@@ -82,7 +84,54 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
+func logPanic(proxyAddr string, msg string) {
+	cfg, err := config.Init(packageVersion)
+	if err != nil {
+		panic("Error initializing config")
+	}
+
+	// Create the client-side proxy.
+	c := &client.Client{
+		// TODO: Next port in case the regular one hasn't been closed properly
+		Addr:         proxyAddr,
+		ReadTimeout:  0, // don't timeout
+		WriteTimeout: 0,
+	}
+	ready := make(chan bool)
+	go c.ListenAndServe(func() { ready <- true })
+	<-ready
+
+	if err := logging.Init(); err != nil {
+		panic("Error initializing logging")
+	}
+
+	ready = logging.Configure(proxyAddr, cfg.CloudConfigCA, cfg.InstanceId, version, revisionDate)
+	<-ready
+
+	log.Error(msg)
+
+	time.Sleep(6 * time.Second)
+	logging.Close()
+}
+
 func main() {
+	// panicwrap works by re-executing the running program (retaining arguments,
+	// environmental variables, etc.) and monitoring the stderr of the program.
+	exitStatus, err := panicwrap.BasicWrap(
+		func(output string) {
+			logPanic("127.0.0.1:18787", output)
```

Comment:
So that's the tradeoff. We can go hard and first try with the proxy. If it fails, do it in the clear. Of course, if the proxy itself is the producer of the panic, we are left out. In any case, this whole thing is a last resort and all errors should be caught before by checks. So if we failed to do so, we better make sure to at least log.
How important is to tunnel loggly messages through our servers?